File: backend/src/main/kotlin/app/cliq/backend/auth/view/TokenResponse.kt
package app.cliq.backend.auth.view

import app.cliq.backend.auth.jwt.TokenPair
import app.cliq.backend.session.Session
import app.cliq.backend.session.view.SessionResponse
import io.swagger.v3.oas.annotations.media.Schema
import java.time.OffsetDateTime

@Schema
class TokenResponse(
    @field:Schema(description = "JWT Access token")
    val accessToken: String,
    @field:Schema(description = "JWT Refresh token")
    val refreshToken: String,
    id: Long,
    name: String? = null,
    lastUsedAt: OffsetDateTime? = null,
    expiresAt: OffsetDateTime,
    createdAt: OffsetDateTime,
) : SessionResponse(id, name, lastUsedAt, expiresAt, createdAt) {
    companion object {
        fun fromTokenPair(tokenPair: TokenPair): TokenResponse =
            fromTokensAndSession(tokenPair.jwt.tokenValue, tokenPair.refreshToken, tokenPair.session)

        fun fromTokensAndSession(
            accessToken: String,
            refreshToken: String,
            session: Session,
        ): TokenResponse =
            TokenResponse(
                accessToken = accessToken,
                refreshToken = refreshToken,
                id = session.id!!,
                name = session.name,
                lastUsedAt = session.lastUsedAt,
                expiresAt = session.expiresAt,
                createdAt = session.createdAt,
            )
    }
}

File: frontend/packages/cliq_api/lib/src/api/responses/token_response.dart
class TokenResponse {
  final int id;
  final String name;
  final DateTime? lastUsedAt;
  final DateTime expiresAt;
  final DateTime createdAt;
  final String accessToken;
  final String refreshToken;

  const TokenResponse({
    required this.id,
    required this.name,
    required this.lastUsedAt,
    required this.expiresAt,
    required this.createdAt,
    required this.accessToken,
    required this.refreshToken,
  });

  factory TokenResponse.fromJson(Map<String, dynamic> json) {
    return TokenResponse(
      id: json['id'] as int,
      name: json['name'] as String,
      lastUsedAt: json['lastUsedAt'] != null
          ? DateTime.parse(json['lastUsedAt'] as String)
          : null,
      expiresAt: DateTime.parse(json['expiresAt'] as String),
      createdAt: DateTime.parse(json['createdAt'] as String),
      accessToken: json['accessToken'] as String,
      refreshToken: json['refreshToken'] as String,
    );
  }
}
